.github/workflows/build.yml

```diff
@@ -19,11 +19,15 @@ jobs:
   macos-build:
     runs-on: macos-latest
     if: "!contains(github.event.pull_request.labels.*.name, 'skip-ci')"
+    defaults:
+      run:
+        working-directory: ./github.com/rook/rook
```

Comment:
@obnoxxx This seems like the secret property that we needed for #14883. Perhaps we can revert that and set the working directory now like this.

Reply:
@travisn  wrote:

> @obnoxxx This seems like the secret property that we needed for #14883. Perhaps we can revert that and set the working directory now like this.

I did experiment with the `working-directory`setting in various updates of that PR but could not get it working.

If we find a more elegant and natural   solution with this, I am of course happy to revert the earlier patch and use this.

Reply:
more precisely, could we split this PR into two since it really has two parts? 

* one PR with the changes to the workflow file
* pne PR with the changes to the script as those are separate and seem to be more controversial.

Reply:
> @travisn wrote:
> 
> > @obnoxxx This seems like the secret property that we needed for #14883. Perhaps we can revert that and set the working directory now like this.
> 
> I did experiment with the `working-directory`setting in various updates of that PR but could not get it working.
> 
> If we find a more elegant and natural solution with this, I am of course happy to revert the earlier patch and use this.

I created a draft PR #15076  to try this.
From the result of the codegen ci workflow, it seems  to be working!